### ISSUE TITLE
[enhancement] Ability to query category children

### ISSUE THREAD

**Issue by tomdavies** [markhuot/craftql#119]
First up, thanks so much for CraftQL, it's super great! I have a feature request (which I'm pretty sure isn't currently possible - apologies if I've missed something).

Essentially just like the structure support discussed in #6, it would be helpful to be able to query the children of a category when querying it. My usecase is for a dynamic category driven navigation.

I.e. I'd like to be able to do something like:

```
categories(level: 1) {
    title
    uri
    children {
    	title
    	uri
    }
}

```

Also, a cheeky extra request to be able to access `level` on `categories` / `categoriesConnection` too (it can be used as an argument but not retrieved AFAICS)

**Comment by markhuot** [comment]
Okay, if you're able to switch to the `dev-master` branch you can try this out. It works exactly as you describe. The following would get you three levels.

```graphql
fragment categoryFields on CategoryInterface {
  id
  title
  level
}

{
  categories(level:1){
    ...categoryFields
    children{
      ...categoryFields
      children{
        ...categoryFields
      }
    }
  }
}
```

In addition you can use this through the `categoriesConnection` field as well,

```graphql
{
  categoriesConnection(level:1){
    edges{
      node{
        id
        title
      }
      children{
        edges{
          node{
            id
            title
          }
        }
      }
    }
  }
}
```

This is only on `dev-master`, at this point. I'll leave this ticket open until I cut a stable tag from it.

**Comment by tomdavies** [comment]
Wow that’s some fast work. Thanks @markhuot - will check it out this weekend!

**Comment by markhuot** [comment]
Closing this issue. The fix is included in version 1.2.1.
